src/psyclone/tests/domain/lfric/lfric_field_test.py

```diff
@@ -233,7 +543,658 @@ def test_field_invoke_uniq_declns_valid_intrinsic():
+        api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=True).create(info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    if annexed:
+        index = 12
+    else:
+        index = 13
+    kernel = schedule.children[index].loop_body[0]
+    # Test w2broken
+    field = kernel.arguments.args[7]
+    assert field.space == 'w2broken'
+    assert field.discontinuous
+    # Test w2vtrace
+    field = kernel.arguments.args[11]
+    assert field.space == 'w2vtrace'
+    assert field.discontinuous
+    # Test wchi
+    field = kernel.arguments.args[4]
+    assert field.space == 'wchi'
+    assert not field.discontinuous
+
+    # 1c) any_discontinuous_space returns true
+    _, info = parse(
+        os.path.join(BASE_PATH,
+                     "1_single_invoke_any_discontinuous_space.f90"),
+        api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=True).create(info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    if annexed:
+        index = 0
+    else:
+        index = 2
+    kernel = schedule.children[index].loop_body[0]
+    field = kernel.arguments.args[0]
+    assert field.space == 'any_discontinuous_space_1'
+    assert field.discontinuous
+
+    # 2) any_space field returns false
+    _, info = parse(os.path.join(BASE_PATH, "11_any_space.f90"),
+                    api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=True).create(info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    if annexed:
+        index = 4
+    else:
+        index = 5
+    kernel = schedule.children[index].loop_body[0]
+    field = kernel.arguments.args[0]
+    assert field.space == 'any_space_1'
+    assert not field.discontinuous
+
+    # 3) Continuous field returns false
+    # 3a) Test w1
+    _, info = parse(os.path.join(BASE_PATH, "1_single_invoke.f90"),
+                    api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=True).create(info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    if annexed:
+        index = 3
+    else:
+        index = 4
+    kernel = schedule.children[index].loop_body[0]
+    field = kernel.arguments.args[1]
+    assert field.space == 'w1'
+    assert not field.discontinuous
+    # 3b) Test w2trace and w2htrace
+    _, info = parse(
+        os.path.join(BASE_PATH,
+                     "1.5.4_single_invoke_write_anyspace_w2trace.f90"),
+        api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=True).create(info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    if annexed:
+        index = 6
+    else:
+        index = 8
+    kernel = schedule.children[index].loop_body[0]
+    # Test w2trace
+    field = kernel.arguments.args[3]
+    assert field.space == 'w2trace'
+    assert not field.discontinuous
+    # Test w2htrace
+    field = kernel.arguments.args[7]
+    assert field.space == 'w2htrace'
+    assert not field.discontinuous
+
+
+# Tests for PSy-layer code generation for field arguments
```

Comment:
So I think this bit ought to have its own test file...

Reply:
Done.